scripts/gitlab/build_and_test.sh

```diff
@@ -112,14 +112,15 @@ then
     echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
 
     # If building, then delete everything first
+    # NOTE: 'cmake --build . -j 12' is an attempt to reduce individual build resources
     rm -rf ${build_dir} 2>/dev/null
     mkdir -p ${build_dir} && cd ${build_dir}
 
     date
     cmake \
       -C ${hostconfig_path} \
       ${project_dir}
-    cmake --build . -j 32
+    cmake --build . -j 12
```

Comment:
I now remember why 32 was chosen:
`build_and_test.sh` runs both on quartz and lassen, and lassen has 64 cores per nodes.
I think we should consider having a value that depends on the hostname. (using a mapping table?).
It seems that 24 would be reasonable on `ruby`. 32 was the usual choice on `lassen`, and 12 worked well on `quartz`.

Reply:
I think that's a great idea!